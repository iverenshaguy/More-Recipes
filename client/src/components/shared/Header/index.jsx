import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Navbar from '../Navbar';
import { logout } from '../../../actions/auth';
import { toggleModal } from '../../../actions/ui';
import { userPropTypes } from '../../../helpers/proptypes';

/**
 * @exports Header
 * @class Header
 * @return {Component} Header
 */
class Header extends Component {
  /**
   * @memberof Header
   * @constructor
   * @returns {JSX} Logged out User
   */
  constructor() {
    super();
    this.logout = this.logout.bind(this);
    this.showAddRecipeModal = this.showAddRecipeModal.bind(this);
  }

  /**
   * @memberof Header
   * @returns {JSX} Logged out User
   */
  logout() {
    this.props.dispatch(logout());
  }

  /**
   * @memberof Header
   * @param {object} e
   * @returns {component} NavbarWrapper
   */
  showAddRecipeModal() {
    this.props.dispatch(toggleModal('addRecipe'));
  }

  /**
   * @memberof Header
   * @returns {JSX} Header
   */
  render() {
    return (
      <header>
        <Navbar

          isAuthenticated={this.props.isAuthenticated}
          logout={this.logout}
          showAddRecipeModal={this.showAddRecipeModal}
          user={this.props.user}
        />
      </header>
    );
  }
}

Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  ...userPropTypes
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  user: state.auth.user
});

export { Header as HeaderComponent };

export default connect(mapStateToProps)(Header);
