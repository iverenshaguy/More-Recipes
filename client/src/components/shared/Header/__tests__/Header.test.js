import React from 'react';
import Header from '../index';

const props = {
  isAuthenticated: false,
  currentLocation: 'auth',
  dispatch: jest.fn()
};

describe('Header', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
