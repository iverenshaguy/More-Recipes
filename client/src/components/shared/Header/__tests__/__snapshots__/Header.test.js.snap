// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <NavbarWrapper
    isAuthenticated={false}
    logout={[Function]}
    type="auth"
  />
</header>
`;
